lib/encrypt/stream.go

```diff
@@ -24,234 +24,195 @@ import (
 
 	"golang.org/x/crypto/blowfish"
 	"golang.org/x/crypto/cast5"
+	"golang.org/x/crypto/chacha20"
 	"golang.org/x/crypto/salsa20/salsa"
 	"golang.org/x/crypto/tea"
 	"golang.org/x/crypto/twofish"
 	"golang.org/x/crypto/xtea"
-
-	"golang.org/x/crypto/chacha20"
 )
 
-// CipherInfo cipher information
-type CipherInfo struct {
-	KeyLen    int
-	IvLen     int
-	NewStream func(key, iv []byte, encrypt bool) (cipher.Stream, error)
-}
-
-var ciphers = map[string]CipherInfo{
-	"aes-128-cfb":     {16, 16, newAesCfbStream},
-	"aes-192-cfb":     {24, 16, newAesCfbStream},
-	"aes-256-cfb":     {32, 16, newAesCfbStream},
-	"aes-128-ctr":     {16, 16, newAesCtrStream},
-	"aes-192-ctr":     {24, 16, newAesCtrStream},
-	"aes-256-ctr":     {32, 16, newAesCtrStream},
-	"aes-128-ofb":     {16, 16, newAesOfbStream},
-	"aes-192-ofb":     {24, 16, newAesOfbStream},
-	"aes-256-ofb":     {32, 16, newAesOfbStream},
-	"des-cfb":         {8, 8, newDesCfbStream},
-	"des-ctr":         {8, 8, newDesCtrStream},
-	"des-ofb":         {8, 8, newDesOfbStream},
-	"blowfish-cfb":    {16, 8, newBlowfishCfbStream},
-	"blowfish-ctr":    {16, 8, newBlowfishCtrStream},
-	"blowfish-ofb":    {16, 8, newBlowfishOfbStream},
-	"cast5-cfb":       {16, 8, newCast5CfbStream},
-	"cast5-ctr":       {16, 8, newCast5CtrStream},
-	"cast5-ofb":       {16, 8, newCast5OfbStream},
-	"twofish-128-cfb": {16, 16, newTwofishCfbStream},
-	"twofish-192-cfb": {24, 16, newTwofishCfbStream},
-	"twofish-256-cfb": {32, 16, newTwofishCfbStream},
-	"twofish-128-ctr": {16, 16, newTwofishCtrStream},
-	"twofish-192-ctr": {24, 16, newTwofishCtrStream},
-	"twofish-256-ctr": {32, 16, newTwofishCtrStream},
-	"twofish-128-ofb": {16, 16, newTwofishOfbStream},
-	"twofish-192-ofb": {24, 16, newTwofishOfbStream},
-	"twofish-256-ofb": {32, 16, newTwofishOfbStream},
-	"tea-cfb":         {16, 8, newTeaCfbStream},
-	"tea-ctr":         {16, 8, newTeaCtrStream},
-	"tea-ofb":         {16, 8, newTeaOfbStream},
-	"xtea-cfb":        {16, 8, newXteaCfbStream},
-	"xtea-ctr":        {16, 8, newXteaCtrStream},
-	"xtea-ofb":        {16, 8, newXteaOfbStream},
-	"rc4-md5":         {16, 16, newRc4Md5Stream},
-	"rc4-md5-6":       {16, 6, newRc4Md5Stream},
-	"chacha20":        {32, 12, newChaCha20Stream},
-	"chacha20-ietf":   {32, 24, newChaCha20IETFStream},
-	"salsa20":         {32, 8, newSalsa20Stream},
-}
-
-// GetCipherInfo 根据方法获得 Cipher information
-func GetCipherInfo(method string) (info CipherInfo, ok bool) {
-	info, ok = ciphers[method]
-	return
-}
-
-// CipherMethods 获取Cipher的所有支持方法
-func CipherMethods() []string {
-	keys := make([]string, 0, len(ciphers))
-	for k := range ciphers {
-		keys = append(keys, k)
-	}
-	return keys
-}
-
-// HasCipherMethod 是否有method方法
-func HasCipherMethod(method string) (ok bool) {
-	_, ok = ciphers[method]
-	return
-}
-
-func newCfbStream(newCipher func(k []byte) (cipher.Block, error), key, iv []byte, encrypt bool) (cipher.Stream, error) {
-	block, err := newCipher(key)
+type KeyIvLen interface {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
exported type KeyIvLen should have comment or be unexported